### PR TITLE
Convert maincontroller to ES6 module

### PR DESCRIPTION
**Pull request by YouKnowBlom**
Changes all `define` to `import` and some minor style changes. Currently there are no exports but this will be pieced together once we're ready to replace requirejs with webpack

### CODE REVIEW

File: src/components/maincontroller.js [hunk 1]
```diff
@@ -1,930 +1,914 @@
...
-    const playbackConfig = new cast.framework.PlaybackConfig();
-    // Set the player to start playback as soon as there are five seconds of
-    // media content buffered. Default is 10.
-    playbackConfig.autoResumeDuration = 5;
-
-    // According to cast docs this should be disabled when not needed
-    cast.framework.CastReceiverContext.getInstance().setLoggerLevel(cast.framework.LoggerLevel.DEBUG);
-
-    var init = function () {
+﻿import { factory as jellyfinActions } from "./jellyfinactions";
```
Comment (dkanada) [inline src/components/maincontroller.js:1]: I would standardize the filenames at some point.

File: src/components/maincontroller.js [hunk 2]
```diff
@@ -1,930 +1,914 @@
...
+    if (data.maxBitrate) {
+        window.MaxBitrate = data.maxBitrate;
+    }
+
+    // Items will have properties - Id, Name, Type, MediaType, IsFolder
+
+    var reportEventType;
+    var systemVolume = window.castReceiverContext.getSystemVolume();
+
+    if (data.command == 'PlayLast' || data.command == 'PlayNext') {
```
Comment (dkanada) [inline src/components/maincontroller.js:242]: Interesting, I forget if we have PlayLast as an action or not.

File: src/components/maincontroller.js [hunk 3]
```diff
@@ -1,930 +1,914 @@
...
-                    }
-                    textTracksManager.setTextTrackStyle(textTrackStyle);
+                if (subtitleAppearance.textBackground === "transparent") {
+                    textTrackStyle.backgroundColor = "#00000000" // RGBA
+                }
+
+                switch (subtitleAppearance.textSize) {
+                    case 'smaller':
+                        textTrackStyle.fontScale = 0.6;
+                        break;
```
Comment (dkanada) [inline src/components/maincontroller.js:790]: This file is really a beast, almost a thosand lines.

File: src/components/maincontroller.js (hunk 1 above)
Comment (YouKnowBlom) [reply src/components/maincontroller.js:1]: Yeah I agree

File: src/components/maincontroller.js (hunk 2 above)
Comment (YouKnowBlom) [reply src/components/maincontroller.js:242]: I'm not quite sure either, I'll have to look into that. Haven't touched maincontroller very much cause I've always wanted to split it up and refactor it asap

File: src/components/maincontroller.js (hunk 3 above)
Comment (YouKnowBlom) [reply src/components/maincontroller.js:790]: Yeah there's so much stuff jammed into it. My plan is to split it up once the ES6 module conversion is done and webpack is set up.

File: src/components/maincontroller.js (hunk 2 above)
Comment (dkanada) [reply src/components/maincontroller.js:242]: We still haven't documented the session controls either, so most of these options are completely unknown.

**Pull request merged**